tests/testthat/test-compute_posterior_intervals.R

```diff
@@ -71,17 +71,17 @@ test_that("compute_posterior_intervals works for SMC", {
 
   pi <- compute_posterior_intervals(mod2)
   expect_equal(pi$parameter, "alpha")
-  expect_equal(pi$median, "2.744")
-  expect_equal(pi$hpdi, "[2.015,3.476]")
-  expect_equal(pi$central_interval, "[2.047,3.530]")
+  expect_equal(pi$median, "2.745")
+  expect_equal(pi$hpdi, "[2.009,3.517]")
+  expect_equal(pi$central_interval, "[2.016,3.519]")
 
   mod3 <- update_mallows(
     mod2,
     new_data = setup_rank_data(potato_visual[10:12, ])
   )
 
   pi <- compute_posterior_intervals(mod3, decimals = 2)
-  expect_equal(pi$hpdi, "[2.18,3.75]")
+  expect_equal(pi$hpdi, "[2.23,3.74]")
   pi <- compute_posterior_intervals(mod3, parameter = "rho")
   expect_equal(pi$hpdi[[20]], "[1,5]")
 
```

Comment:
This is alright, but for future-proofing, perhaps we could compare numbers with a high tolerance (like 0.1)?

Reply:
I agree. I just have to figure out the random number generation stuff on my M1 mac. On all other platforms the results are reproduced, but M1 mac seems to work differently, even though I use Rcpp's random number generators which are designed to by platform independent. I don't know if it's the RNGs fault or if there are differences in floating point representations or something like that which causes the input to the functions for generating random numbers to differ. I think my next laptop will run Linux.